swap_meet/vendor.py

```diff
@@ -0,0 +1,71 @@
+
+class Vendor:
+    def __init__(self, inventory=None): 
+        
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item): 
+        if item not in self.inventory:
+            return False 
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+    
+    def get_by_category(self, category):
+        matching_items = [] 
+        for item in self.inventory:
+            if category == item.category:
+                matching_items.append(item)
+
+        return  matching_items
+    
+    def swap_items(self, friend, my_item, friends_item):
+        if my_item in self.inventory and friends_item in friend.inventory:
+            # friend.add(self.remove(my_item))
+            # self.add(friend.remove(friends_item))
+            self.remove(my_item)
+            friend.add(my_item)
+            friend.remove(friends_item)
+            self.add(friends_item)                 
+            return True
+        else:
+            return False
```

Comment:
Consider reorganizing like this to deal with the edge case first and move the main logic of the function out from being nested.

```suggestion
        if my_item not in self.inventory or friends_item not in friend.inventory:
            return False
            
        self.remove(my_item)
        friend.add(my_item)
        friend.remove(friends_item)
        self.add(friends_item)                 
        return True
```